content/200-concepts/100-components/02-prisma-client/260-client-extensions/130-result.mdx

```diff
@@ -1,16 +1,159 @@
 ---
-title: 'Add custom results objects to Prisma Client'
-metaTitle: 'Prisma Client extensions: result (Preview)'
+title: 'Add custom fields and methods to query results'
+metaTitle: 'Prisma Client extensions: result component (Preview)'
 metaDescription: 'Extend the functionality of Prisma Client'
 tocDepth: 3
 ---
 
 <TopBlock>
 
-<Admonition type="warning">
+From version 4.7.0, you can use the `result` [Prisma Client extensions](/concepts/components/prisma-client/client-extensions) component to add custom fields and methods to query results.
 
-We plan to add `result` extensions to a future version of Prisma. This page is a placeholder.
+</TopBlock>
 
-</Admonition>
+## Enable the preview feature
 
-</TopBlock>
+Before you create Prisma Client extensions, you must enable the `clientExtensions` feature flag in the `generator` block of your `schema.prisma` file, as follows:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+```
+
+## Add custom fields or methods to query results
+
+Use the `$extends` [client-level method](/reference/api-reference/prisma-client-reference#client-methods) to create an _extended client_. An extended client is a variant of the standard Prisma Client that is wrapped by one or more extensions.
+
+Use the `result` extension component to add custom fields and methods to query results.
+
+To add a custom [field](#add-a-custom-field-to-the-results-object) or [method](#add-a-custom-method-to-the-results-object) to query results, use the following structure. In this example, we add a custom field to the `user` model.
+
+```ts
+const xprisma = prisma.$extends({
+  name?: 'name',
+  result?: {
+    user: {     // in this case, we extend the `user` model
+      needs: { ... }
+      fields: { ... }
+    },
+  },
+});
+```
+
+** do we need a compute statement here? Is compute always a component? if so mention it 1. above and 2. in the parameters below**
+
+The parameters are as follows:
+
+- `name`: (optional) specifies a name for the extension that appears in error logs.
+- `result`: (optional) defines new fields and methods to the query results.
+- `needs`: (required) an object which describes the dependencies of the result field.
+
+** I think we remove fields **
+
+- `fields`: (required) an object that contains methods that define how the virtual field is computed when it is accessed. Properties declared in `fields` must also be declared in `needs`.
+- `return`: the value of the computed field.
```

Comment:
We don't include `return` in the code example above

Reply:
I think this section is completely outdated. `fields` does not exist anymore and I am not sure what `return` refers to here - I don't recall anything by that name at any stage.